manim/animation/transform.py

```diff
@@ -38,26 +48,30 @@
 from ..utils.paths import path_along_arc, straight_path
 from ..utils.rate_functions import smooth, squish_rate_func
 
-if typing.TYPE_CHECKING:
+if TYPE_CHECKING:
     from ..scene.scene import Scene
 
 
 class Transform(Animation):
     def __init__(
         self,
-        mobject: Mobject,
-        target_mobject: typing.Optional[Mobject] = None,
-        path_func: typing.Optional[typing.Callable] = None,
+        mobject: Optional[Mobject],
+        target_mobject: Optional[Mobject] = None,
+        path_func: Optional[Callable] = None,
         path_arc: float = 0,
         path_arc_axis: np.ndarray = OUT,
         replace_mobject_with_target_in_scene: bool = False,
         **kwargs,
     ) -> None:
-        self.path_arc = path_arc
-        self.path_func = path_func
-        self.path_arc_axis = path_arc_axis
-        self.replace_mobject_with_target_in_scene = replace_mobject_with_target_in_scene
-        self.target_mobject = target_mobject
+        self.path_arc: float = path_arc
+        self.path_func: Optional[Callable] = path_func
+        self.path_arc_axis: np.ndarray = path_arc_axis
+        self.replace_mobject_with_target_in_scene: bool = (
+            replace_mobject_with_target_in_scene
+        )
+        self.target_mobject: Mobject = (
+            target_mobject if target_mobject is not None else Mobject()
```

Comment:
The target_mobject keyword argument is optional and always has been. Animations often don't need you to specify the target_mobject, because they generate their own target using self.create_target()

What this has changed is that self.target_mobject is not optional and if the keyword argument is set to None, then self.target_mobject is initialised to the empty Mobject() until and unless overwritten by self.create_target()